File: set3/7/main.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    int N, K; cin >> N >> K;
    vector<int> arr(N); 
    for(int i = 0; i < N; ++i) cin >> arr[i];

    int erased = 0, answer = 0;;
    for(int i = 0, R = 0; i < N; i++) {
        while(R < N) {
            if(arr[R] % 2 == 0) R++;
            else {
                if(erased == K) break;
                erased++;
                R++;
            }
        } 
        answer = max(answer, R - i - erased);
        if(arr[i] % 2) erased--;
    }
    cout << answer;

    return 0;
}

File: set3/7/main.py
import sys

def input():
    return sys.stdin.readline().rstrip()

N, K = map(int, input().split())
arr = list(map(int, input().split()))

erased, answer = 0, 0
R = 0
for i in range(N):
    while R < N:
        if arr[R] % 2 == 0: R += 1
        else:
            if erased == K:
                break
            erased += 1
            R += 1
    answer = max(answer, R - i - erased)
    if arr[i] % 2 == 1:
        erased -= 1
print(answer)